Player.cpp

```diff
@@ -0,0 +1,37 @@
+#include "Player.h"
+
+Player::Player(std::string name, float hp, float dmg) : name(name),hp(hp),dmg(dmg){}
+
+void Player::GetPlayerStatus()
+{
+	std::cout << name << ": HP: "<< this->hp<<", DMG: "<<this->dmg<< "\n";
+}
+
+const std::string Player::GetName()
+{
+	return this->name;
+}
```

Comment:
Szerintem itt nem azt csinálod, amit szerettél volna, mert a visszaadott érték amúgy is másolata az eredetinek, és őszintén, senkit nem érdekel, ha valaki ezt a másolatot átalakítja. Itt az objektum adattagjának értékét kellene megvédeni, de erre meg az így elhelyezett const nem alkalmas. Szerintem Te azt a sor végére akartad írni igazából, de mielőtt bármit átalakítanál, olvasd el pl. ezt:
https://www.linuxtopia.org/online_books/programming_books/thinking_in_c++/Chapter08_014.html